providers/darwin/process_darwin_amd64.go

```diff
@@ -40,7 +40,38 @@ import (
 //go:generate sh -c "go tool cgo -godefs defs_darwin.go > ztypes_darwin_amd64.go"
 
 func (s darwinSystem) Processes() ([]types.Process, error) {
-	return nil, nil
+	n, err := C.proc_listallpids(nil, 0)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting process count from proc_listallpids (n = %v)", n)
+	} else if n <= 0 {
+		return nil, errors.Errorf("proc_listallpids returned %v", n)
+	}
+
+	var pid C.int
+	bufsize := n * C.int(unsafe.Sizeof(pid))
+	buf := make([]byte, bufsize)
+	n, err = C.proc_listallpids(unsafe.Pointer(&buf[0]), bufsize)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting processes from proc_listallpids (n = %v)", n)
+	} else if n <= 0 {
+		return nil, errors.Errorf("proc_listallpids returned %v", n)
+	}
+
+	bbuf := bytes.NewBuffer(buf)
+	processes := make([]types.Process, 0, n)
+	for i := 0; i < int(n); i++ {
+		err = binary.Read(bbuf, binary.LittleEndian, &pid)
```

Comment:
This is another of those places where we could benefit from a binary.MachineEndian.